app/Http/Controllers/ProductsController.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Products;
+use Carbon\Carbon;
+use App\Helpers\ResponseHelper;
+
+
+class ProductsController extends Controller
+{
+    public function getData(Request $req)
+    {
+
+        $products = Products::all();
+
+        if (!$req->query()) {
+
+            return response()->json(ResponseHelper::sendResponse(true, 200, "Data Fetched Successfully", $products));
+        }
+
+        //initializing default values
+        $startDate = Carbon::createFromFormat('Y-m-d', '1999-07-01')->startOfDay();
```

Comment:
no need to initialize these variables as if we are not using they will consume space in my memory